addons/sale/models/sale_order.py

```diff
@@ -528,6 +533,36 @@ def _compute_amounts(self):
             order.amount_tax = tax_totals['tax_amount_currency']
             order.amount_total = tax_totals['total_amount_currency']
 
+    @api.depends(
+        'order_line.qty_delivered',
+        'order_line.product_uom_qty',
+    )
+    def _compute_is_unfulfilled(self):
+        for order in self:
+            order.is_unfulfilled = (
+                any(
+                    line.qty_delivered < line.product_uom_qty
+                    for line in order.order_line
+                )
+                and order.state == 'sale'
+            )
+
+    def _search_is_unfulfilled(self, operator, value):
+        if operator not in ('=', '!='):
+            return NotImplemented
+
+        operator = 'any' if operator == '=' else 'not any'
+
+        line_domain = Domain.custom(
+            to_sql=lambda model, alias, query: SQL(
+                '%s < %s',
+                model._field_to_sql(alias, 'qty_delivered', query),
+                model._field_to_sql(alias, 'product_uom_qty', query),
+            ),
```

Comment:
sometimes I get the error in local
`SaleOrder._search_is_unfulfilled.<locals>.<lambda>() missing 2 required positional arguments: 'alias' and 'query'` Not sure why 🤔 but you can do this in simpler way like this
```suggestion
            to_sql=lambda table: SQL('%s < %s', table.qty_delivered, table.product_uom_qty),
```
(Took reference from purchase)
WDYT?

Reply:
But we could avoid calling `_field_to_sql` and make it simpler [for ref](https://github.com/odoo/odoo/blob/master/addons/purchase/models/purchase_order.py#L371)